core/transaction_pool_test.go

```diff
@@ -238,3 +238,15 @@ func TestNonceRecovery(t *testing.T) {
 		t.Errorf("expected nonce to be %d, got %d", n+1, fn)
 	}
 }
+
+func TestRemovedTxEvent(t *testing.T) {
+	pool, key := setupTxPool()
+	tx := transaction(0, big.NewInt(1000000), key)
+	from, _ := tx.From()
+	pool.currentState().AddBalance(from, big.NewInt(1000000000000))
+	pool.eventMux.Post(RemovedTransactionEvent{types.Transactions{tx}})
+	pool.eventMux.Post(ChainHeadEvent{nil})
```

Comment:
@fjl I had to post this event in order for the `transaction_pool` to receive the `RemovedTransactionEvent`. The `RemovedTransactionEvent` wasn't fired until this one was posted.`Post` is supposed to be blocking, iirc. Any clue?